nimare/diagnostics.py

```diff
@@ -0,0 +1,236 @@
+        stat_values = result.get_map(target_value_map, return_type="array")
+
+        # Use study IDs in inputs_ instead of dataset, because we don't want to try fitting the
+        # estimator to a study that might have been filtered out by the estimator's criteria.
+        meta_ids = estimator.inputs_["id"]
+        rows = ["Center of Mass"] + list(meta_ids)
+
+        # Let's label the clusters in the thresholded map so we can use it as a NiftiLabelsMasker
+        # This won't work when the Estimator's masker isn't a NiftiMasker... :(
+        conn = np.zeros((3, 3, 3), int)
+        conn[:, :, 1] = 1
+        conn[:, 1, :] = 1
+        conn[1, :, :] = 1
+        labeled_cluster_arr, n_clusters = ndimage.measurements.label(thresh_arr, conn)
+        labeled_cluster_img = nib.Nifti1Image(
+            labeled_cluster_arr,
+            affine=target_img.affine,
+            header=target_img.header,
+        )
+
+        if n_clusters == 0:
+            LGR.warning("No clusters found")
+            contribution_table = pd.DataFrame(index=rows)
+            return contribution_table, labeled_cluster_img
+
+        # Identify center of mass for each cluster
+        # This COM may fall outside the cluster, but it is a useful heuristic for identifying them
+        cluster_ids = list(range(1, n_clusters + 1))
+        cluster_coms = ndimage.center_of_mass(
+            labeled_cluster_arr,
+            labeled_cluster_arr,
+            cluster_ids,
+        )
+        cluster_coms = np.array(cluster_coms)
+        cluster_coms = vox2mm(cluster_coms, target_img.affine)
+
+        cluster_com_strs = []
+        for i_peak in range(len(cluster_ids)):
+            x, y, z = cluster_coms[i_peak, :].astype(int)
+            xyz_str = f"({x}, {y}, {z})"
+            cluster_com_strs.append(xyz_str)
+
+        # Mask using a labels masker, so that we can easily get the mean value for each cluster
+        cluster_masker = input_data.NiftiLabelsMasker(labeled_cluster_img)
+        cluster_masker.fit(labeled_cluster_img)
+
+        # Create empty contribution table
+        contribution_table = pd.DataFrame(index=rows, columns=cluster_ids)
+        contribution_table.index.name = "Cluster ID"
+        contribution_table.loc["Center of Mass"] = cluster_com_strs
+
+        # For the private method that does the work of the analysis, we have a lot of constant
+        # parameters, and only one that varies (the ID of the study being removed),
+        # so we use functools.partial.
+        transform_method = partial(
+            self._transform,
+            all_ids=meta_ids,
+            dset=dset,
+            estimator=estimator,
+            target_value_map=target_value_map,
+            stat_values=stat_values,
+            original_masker=original_masker,
+            cluster_masker=cluster_masker,
+        )
+        with Pool(self.n_cores) as p:
+            jackknife_results = list(tqdm(p.imap(transform_method, meta_ids), total=len(meta_ids)))
```

Comment:
This seems like a much cleaner approach than what I've been doing for Monte Carlo FWE correction in the CBMA classes. However, one thing I'm not sure about is if using `Pool` with one process is the same as running the same function without multiprocessing at all. @jdkent do you know?